config/nginx.conf.erb

```diff
@@ -28,6 +28,8 @@ http {
 	client_body_timeout 30;
 	client_max_body_size 50m;
 
+	limit_req_zone $binary_remote_addr zone=publish:1m rate=1r/s;
```

Comment:
I'm not certain, but I think on Heroku this will be an IP address from their routing infrastructure.  We can instead use $http_x_forwarded_for but this takes up more memory than the binary representation and I'm not 100% how Heroku handles this.  By that I mean if a malicious user includes their own version of this header then Heroku may append or prepend the users real IP address, but effectively allow them to change the final value in this header with each request.

We might be able to use the [realip](http://nginx.org/en/docs/http/ngx_http_realip_module.html) module to properly set $binary_remote_addr, but it looks like that isn't compiled in by default and I haven't checked to see if our nginx buildpack includes `--with-http_realip_module`.

Reply:
Yeah I think using x-forwarded-for makes sense. I'm not worried about the memory usage too much, we aren't going to get enough requests here for it to really matter, and we can easily up this to whatever size we need in the future if it's a problem.

Reply:
Okay, sounds good to me.

Reply:
Holding off for now since I want to change our nginx buildpack and the heroku maintained one does have the realip module which I would just generally like to use